### PR TITLE
Habit Digest snap banner

### PR DESCRIPTION
**Pull request by katebee**
## What does this change?

Adds an **inactive** test (currently off and at zero audience) for the Habit Digest promo. This test will show users a banner offering a curated digest.

The banner co-opts the Membership Message banner to display one of three variant messages to the visitor (see screenshots). It has a lower priority than the Breaking News banner, and will not appear again once a visitor has dismissed it.
- Users will only see one snap variant
- Users will only see the snap once
- Breaking news gets priority

The sign-up button is a false door / 404 test that will open a info page.
## What is the value of this and can you measure success?
- Discover if there is demand for various Guardian digest offerings
- See which version of the copy generates greatest engagement

Ideally, visitors will click on the CTA and demonstrate demand for the feature.
## Does this affect other platforms - Amp, Apps, etc?

No
## Screenshots

picture 126
## For a subsequent PR:
- ~~Check the links in the [banner variants](https://github.com/guardian/frontend/blob/2920e52b7a8c1cbb1a6204f9964211f6a846103c/static/src/javascripts/projects/common/modules/experiments/tests/habit-forming-digest-promo.js#L96): are they correct and do the landing pages exist?~~ **done**
- Run the test (turn AB switch on and set audience percentage)
## Request for comment

@joelochlann

### CODE REVIEW

File: static/src/javascripts/projects/common/modules/experiments/tests/habit-forming-digest-promo.js [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+
+        var defaultData = {
+            arrowRight: svgs('arrowRight')
+        };
+
+        // check if the user is one of the target audience
+        function isInfrequentVisitor() {
+            if (storage.local.isStorageAvailable()) {
+                var alreadyVisited = storage.local.get('gu.alreadyVisited') || 0;
+                if (alreadyVisited > 3 && !userFeatures.isPayingMember()) {
```
Comment (joelochlann) [inline static/src/javascripts/projects/common/modules/experiments/tests/habit-forming-digest-promo.js]: Hmmmm... this seems to be doing the opposite of what it says! It's looking for people who have visited the site more than 3 times.

Comment (joelochlann) [inline static/src/javascripts/projects/common/modules/experiments/tests/habit-forming-digest-promo.js]: or maybe it's the function name that's misleading

Comment (janua) [inline static/src/javascripts/projects/common/modules/experiments/tests/habit-forming-digest-promo.js]: @joelochlann I have also read it as the opposite! The double negative is confusing.

Comment (joelochlann) [inline static/src/javascripts/projects/common/modules/experiments/tests/habit-forming-digest-promo.js]: from Monica:

> Re audience, we will reverse analyse this afterwards, so of the people who see & interact with this, which fall into the non-regular / regular buckets of users etc as there is no good way to target those audiences at the moment.

so all we should be doing is:
* ensuring users only see the snap once
* ensuring we don't conflict with the membership snap

so it looks to me like this whole condition is irrelevant.

**Comment by SiAdcock** [comment]
LGTM 👍

**Pull request merged**